src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -53,16 +55,63 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
                     continue;
                 }
 
-                if (node.TagHelper.IsBindTagHelper() && node.AttributeName.StartsWith("bind"))
+                if (node.TagHelper.IsBindTagHelper() && node.AttributeName.StartsWith("bind") && !node.IsParameterMatch)
                 {
-                    // Workaround for https://github.com/aspnet/Blazor/issues/703
-                    var rewritten = RewriteUsage(reference.Parent, node);
-                    reference.Remove();
+                    bindEntries[node.AttributeName] = new BindEntry(reference);
+                }
+            }
+
+            // Now collect all the parameterized attributes and store them along with their corresponding bind or bind-* attributes.
+            for (var i = 0; i < references.Count; i++)
+            {
+                var reference = references[i];
+                var node = (TagHelperPropertyIntermediateNode)reference.Node;
+
+                if (!reference.Parent.Children.Contains(node))
+                {
+                    // This node was removed as a duplicate, skip it.
+                    continue;
+                }
 
-                    for (var j = 0; j < rewritten.Length; j++)
+                if (node.TagHelper.IsBindTagHelper() && node.AttributeName.StartsWith("bind") && node.IsParameterMatch)
+                {
+                    var originalAttributeName = node.AttributeName.Split(':')[0];
+                    if (!bindEntries.ContainsKey(originalAttributeName))
+                    {
+                        // There is no corresponding bind node. Add a diagnostic and move on.
+                        reference.Parent.Diagnostics.Add(ComponentDiagnosticFactory.CreateBindAttributeParameter_MissingBind(
+                            node.Source,
+                            node.AttributeName));
+                    }
+                    else if (node.BoundAttributeParameter.Name == "event")
+                    {
+                        bindEntries[originalAttributeName].BindEventNode = node;
+                    }
+                    else if (node.BoundAttributeParameter.Name == "format")
+                    {
+                        bindEntries[originalAttributeName].BindFormatNode = node;
+                    }
+                    else
                     {
-                        reference.Parent.Children.Add(rewritten[j]);
+                        // Unsupported bind attribute parameter.
```

Comment:
Should this be an error?

Reply:
Technically this case will never be hit. Because for cases like bind:unknown='...', this will never be a tag helper attribute so will never be a `TagHelperPropertyIntermediateNode`. Aka it will never hit this code path.